x/poolmanager/types/msg_swap.go

```diff
@@ -0,0 +1,48 @@
+package types
+
+// SwapMsg defines a simple interface for getting the token denoms on a swap message route.
+type SwapMsgRoute interface {
+	TokenInDenom() string
+	TokenOutDenom() string
+	TokenDenomsOnPath() []string
+}
+
+var (
```

Comment:
I am aware I am commenting on a merged PR, and that there's ongoing work https://github.com/osmosis-labs/osmosis/pull/6811 (yet it seems it's not touching this part).

But out of curiosity, shouldn't arb detection also deal with `MsgSplitRouteSwapExactAmount[In|Out]`

example: https://www.mintscan.io/osmosis/tx/FF071BB716F3F829A9E4AC21C8CBFD1011E82EDF8F9B1088F6DB2AED552E1BF0?height=12174133

@ValarDragon